### PR TITLE
Update ASP.NET packages to version 24686

### PR DESCRIPTION
**Pull request by mlorbetske**
Updates ASP.NET package versions
Switches ASP.NET feeds to release
Updates test SDK version
Adds a fix for switches not being displayed in help if it has no short form
Fixes an issue where files at the root of a template could not be excluded

### CODE REVIEW

File: NuGet.Config [hunk 1]
```diff
@@ -5,7 +5,7 @@
     <clear />
     <add key="templating" value="https://dotnet.myget.org/F/templating/api/v3/index.json" />
     <add key="dotnet-core" value="https://dotnet.myget.org/F/dotnet-core/api/v3/index.json" />
-    <add key="aspnetcore-dev" value="https://dotnet.myget.org/F/aspnetcore-dev/api/v3/index.json" />
+    <add key="aspnetcore-release" value="https://dotnet.myget.org/F/aspnetcore-release/api/v3/index.json" />
```
Comment (livarcocc) [inline NuGet.Config]: Should we add this one to our mirror as well?

It would be nice if we stopped changing these.

Comment (mlorbetske) [inline NuGet.Config]: Yep, we should mirror this one. They just branched off for preview 1, hence the change. They toggle between two depending on the current point in the release cycle.

**Comment by mlorbetske** [comment]
@dotnet-bot test Ubuntu x64 Release Build please

**Comment by mlorbetske** [comment]
@dotnet-bot test Debian8.2 x64 Debug Build please

**Comment by mlorbetske** [comment]
@dotnet-bot test Ubuntu x64 Release Build please

**Comment by livarcocc** [comment]
@MattGertz for approval.

**Comment by MattGertz** [comment]
Approved pending test completion.

**Comment by mlorbetske** [comment]
Looks like the Ubuntu x64 Release build is failing with the error below, any ideas? @livarcocc 

```
                   Task "DotNetDebTool" (TaskId:2936)
                     Task Parameter:ToolPath=/opt/code/.dotnet_stage0/x64 (TaskId:2936)
                     Task Parameter:InputDirectory=/opt/code/artifacts/ubuntu.14.04-x64/intermediate/debian/sdk/debianLayoutDirectory (TaskId:2936)
                     Task Parameter:OutputDirectory=/opt/code/artifacts/ubuntu.14.04-x64/intermediate/debian/sdk/deb-tool-output (TaskId:2936)
                     Task Parameter:PackageName=dotnet-dev-2.0.0-preview1-005895 (TaskId:2936)
                     Task Parameter:PackageVersion=2.0.0-preview1-005895 (TaskId:2936)
                     Task Parameter:WorkingDirectory=/opt/code/artifacts/ubuntu.14.04-x64/intermediate/dotnet-deb-tool-consumer.csproj (TaskId:2936)
                     /opt/code/artifacts/ubuntu.14.04-x64/intermediate/dotnet-deb-tool-consumer.csproj> [DotNetTool] deb-tool -i /opt/code/artifacts/ubuntu.14.04-x64/intermediate/debian/sdk/debianLayoutDirectory -o /opt/code/artifacts/ubuntu.14.04-x64/intermediate/debian/sdk/deb-tool-output -n dotnet-dev-2.0.0-preview1-005895 -v 2.0.0-preview1-005895 (TaskId:2936)
                     /opt/code/artifacts/ubuntu.14.04-x64/intermediate/dotnet-deb-tool-consumer.csproj> /opt/code/.dotnet_stage0/x64/dotnet deb-tool -i /opt/code/artifacts/ubuntu.14.04-x64/intermediate/debian/sdk/debianLayoutDirectory -o /opt/code/artifacts/ubuntu.14.04-x64/intermediate/debian/sdk/deb-tool-output -n dotnet-dev-2.0.0-preview1-005895 -v 2.0.0-preview1-005895  (TaskId:2936)
                     It was not possible to find any compatible framework version (TaskId:2936)
                     The specified framework 'Microsoft.NETCore.App', version '1.0.0' was not found. (TaskId:2936)
                       - Check application dependencies and target a framework version installed at: (TaskId:2936)
                           / (TaskId:2936)
                       - Alternatively, install the framework version '1.0.0'. (TaskId:2936)
07:24:19.125     1>/opt/code/build/package/Installer.DEB.proj(55,7): error MSB6006: "dotnet" exited with code 131. [/opt/code/build.proj]
                   Done executing task "DotNetDebTool" -- FAILED. (TaskId:2936)
07:24:19.126     1>Done building target "BuildSdkDeb" in project "build.proj" -- FAILED.: (TargetId:366)
07:24:19.126     1>Done Building Project "/opt/code/build.proj" (default targets) -- FAILED.
```

**Comment by livarcocc** [comment]
@dotnet-bot test Ubuntu x64 Release Build

**Pull request merged**